packages/itwin/reports-config-widget/src/widget/components/ReportHorizontalTile.tsx

```diff
@@ -0,0 +1,130 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { useEffect, useState } from "react";
+import React from "react";
+import type BulkExtractor from "./BulkExtractor";
+import { ExtractionStates, ExtractionStatus } from "./ExtractionStatus";
+import type { BeEvent } from "@itwin/core-bentley";
+import { STATUS_CHECK_INTERVAL } from "./Constants";
+import type { Report } from "@itwin/insights-client";
+import { ReportsConfigWidget } from "../../ReportsConfigWidget";
+import {
+  DropdownMenu,
+  IconButton,
+  MenuItem,
+} from "@itwin/itwinui-react";
+import {
+  SvgDelete,
+  SvgEdit,
+  SvgMore,
+} from "@itwin/itwinui-icons-react";
+import { HorizontalTile } from "./HorizontalTile";
+
+export interface ReportHorizontalTileProps {
+  onClickTitle: (e: any) => void;
+  selected: boolean;
+  setSelectedReports: (value: React.SetStateAction<Report[]>) => void;
```

Comment:
Selection should be handled by the parent component. Change this to `onSelectionChange`.
First parameter - reportId.
Second - boolean for selected or deselected.
Third - optional boolean, which would be set when second parameter is true and means that the selected report should be appended or replace the whole list.

Reply:
Added `onSelectionChange` instead of `setSelectedReports`. `onSelectedChange` accepts two parameters: `reportId` and `controlPressed` - boolean. The event is then handled at the parent component. Is that correct?